### PR TITLE
Setup user role

### PR DESCRIPTION
**Pull request by rhnonose**
- [x] Migrate to Guardian 1.0
 - [x] Refactor routes to separate public and private in the `router.ex` file.

### CODE REVIEW

File: lib/re_web/controllers/user_controller.ex [hunk 1]
```diff
@@ -34,7 +31,7 @@ defmodule ReWeb.UserController do
   def create(conn, %{"user" => user_params}) do
     case Auth.register(user_params) do
       {:ok, user} ->
-        {:ok, jwt, _full_claims} = user |> Guardian.encode_and_sign(:token)
+        {:ok, jwt, _full_claims} = user |> Guardian.encode_and_sign()
```
Comment (gusaiani) [inline lib/re_web/controllers/user_controller.ex]: Use a function call when a pipeline is only one function long

File: lib/re_web/controllers/session_controller.ex [hunk 2]
```diff
@@ -2,11 +2,12 @@ defmodule ReWeb.SessionController do
   use ReWeb, :controller
 
   alias Re.Accounts.Auth
+  alias ReWeb.Guardian
 
   def create(conn, params) do
     case Auth.find_user_and_check_password(params) do
       {:ok, user} ->
-        {:ok, jwt, _full_claims} = user |> Guardian.encode_and_sign(:token)
+        {:ok, jwt, _full_claims} = user |> Guardian.encode_and_sign()
```
Comment (gusaiani) [inline lib/re_web/controllers/session_controller.ex]: Use a function call when a pipeline is only one function long

File: lib/re_web/auth/guardian_pipeline.ex [hunk 3]
```diff
@@ -0,0 +1,12 @@
+defmodule ReWeb.GuardianPipeline do
```
Comment (gusaiani) [inline lib/re_web/auth/guardian_pipeline.ex]: Modules should have a @moduledoc tag.

File: lib/re_web/auth/auth_error_handler.ex [hunk 4]
```diff
@@ -0,0 +1,16 @@
+defmodule ReWeb.Guardian.AuthErrorHandler do
```
Comment (gusaiani) [inline lib/re_web/auth/auth_error_handler.ex]: Modules should have a @moduledoc tag.

**Comment by gusaiani** [comment]
Ebert has finished reviewing this Pull Request and has found:

* 5 possible new issues (including those that may have been commented here).
* 4 fixed issues! 🎉

But beware that this branch is **[2 commits behind the `em-casa:master` branch](https://github.com/em-casa/re/compare/em-casa:nonose/user_roles...em-casa:master)**, and a review of an up to date branch would produce more accurate results.

You can see more details about this review at https://ebertapp.io/github/em-casa/re/pulls/48.

**Comment by gusaiani** [comment]
Ebert has finished reviewing this Pull Request and has found:

* 1 possible new issue (including those that may have been commented here).
* 4 fixed issues! 🎉

But beware that this branch is **[2 commits behind the `em-casa:master` branch](https://github.com/em-casa/re/compare/em-casa:nonose/user_roles...em-casa:master)**, and a review of an up to date branch would produce more accurate results.

You can see more details about this review at https://ebertapp.io/github/em-casa/re/pulls/48.

**Comment by gusaiani** [comment]
Ebert has finished reviewing this Pull Request and has found:

* 1 possible new issue (including those that may have been commented here).
* 4 fixed issues! 🎉

You can see more details about this review at https://ebertapp.io/github/em-casa/re/pulls/48.

**Comment by gusaiani** [comment]
Ebert has finished reviewing this Pull Request and has found:

* 6 fixed issues! 🎉

You can see more details about this review at https://ebertapp.io/github/em-casa/re/pulls/48.

File: priv/repo/migrations/20180108183622_add_user_role.exs [hunk 5]
```diff
@@ -0,0 +1,9 @@
+defmodule Re.Repo.Migrations.AddUserRole do
+  use Ecto.Migration
+
+  def change do
+    alter table(:users) do
+      add :role, :string, default: "admin"
```
Comment (gusaiani) [inline priv/repo/migrations/20180108183622_add_user_role.exs]: @rhnonose, o default não deveria ser "admin" desde o início, para a gente não correr risco de segurança no futuro.

Comment (rhnonose) [reply priv/repo/migrations/20180108183622_add_user_role.exs]: Verdade. Eu coloquei assim pra settar como `admin` os usuários existentes.
Vou fazer um update pra settar nos existentes e tirar do default.

**Pull request merged**